core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -434,6 +434,11 @@ public StructLike partition() {
     return partitionData;
   }
 
+  public void setPartitionData(PartitionData partitionData) {
```

Comment:
I don't think this is necessary, we should avoid it.

Reply:
Agreed. @rahil-c could we just reuse the existing constructor and copy over everything correctly with the new partition data during the binding? We should avoid exposing unnecessary public APIs when possibke

Reply:
Ah BaseFile isn't public itself but still, think we should just avoid the extra method since it's possible to do everything via the constructor.